test/CodeGen/RISCV/calu.ll

```diff
@@ -0,0 +1,138 @@
+; RUN: llc -mtriple=riscv32imac -verify-machineinstrs < %s | FileCheck %s
+
+; Function Attrs: noinline nounwind optnone
+define i32 @ci() #0 {
+; CHECK: c.li    a0, 0
+  ret i32 0
+}
+
+; Function Attrs: noinline nounwind optnone
+define i32 @caddi(i32 signext) #0 {
+  %2 = alloca i32, align 4
```

Comment:
Why not write the minimal IR to generate the correct pattern? More complex tests are always welcome, but compiler-generated unoptimised code probably just obfuscates things here

Reply:
Update calu.ll with much clean and clear IR.